---
title: "Cost of a spore"
author: "Jay T. Lennon"
date: "`r format(Sys.time(), '%d %B, %Y')`"
header-includes:
   - \usepackage{array}
output: pdf_document
geometry: margin=2.54cm
---

# Overview
Estimate cost of a cell and a spore from replication, transcription, and translation

# Set working directory and load packages

```{r}
rm(list = ls())
getwd()
setwd("~/GitHub/SporeCost")
require("seqinr")
```

# I. Cost at the gene level following Lynch and Marinov (2015)

```{r}
# Plasmid = https://www.ncbi.nlm.nih.gov/nuccore/CP020103, 84,215 bp

plasmid.raw <- read.fasta(file = "data/replication/Bs3610plasmid.fa")
length(plasmid.raw) # should be 1 sequence
plasmid <- plasmid.raw[[1]]
plasmid.nucleotides <- count(plasmid,1)
plasmid.GC <- GC(plasmid)
plasmid.A <- plasmid.nucleotides[1]
plasmid.C <- plasmid.nucleotides[2]
plasmid.G <- plasmid.nucleotides[3]
plasmid.T <- plasmid.nucleotides[4]
plasmid.genome <- sum(plasmid.A, plasmid.C, plasmid.G, plasmid.T)
```

```{r}
# Chromosome = https://www.ncbi.nlm.nih.gov/nuccore/CP020102, 4,215,607 bp
chrom.raw <- read.fasta(file = "data/replication/Bs3610chromosome.fa")
length(chrom.raw) # should be 1 sequence
chrom <- chrom.raw[[1]]
chrom.nucleotides <- count(chrom,1)
chrom.GC <- GC(chrom)
chrom.A <- chrom.nucleotides[1]
chrom.C <- chrom.nucleotides[2]
chrom.G <- chrom.nucleotides[3]
chrom.T <- chrom.nucleotides[4]
chrom.genome <- sum(chrom.A, chrom.C, chrom.G, chrom.T)
```

# 1) Costs of nucelotide synthesis

```{r}
# 1) Nucleotide synthesis: convert genomic info above into costs of biosynthesizing new nucelotides
# Treat costs in units of phosphate bonds hydrolyzed, denoted as P

dATP <- 51.7
dCTP <- 48.7
dGTP <- 52.7
dTTP <- 47.7

plasmid.synth.P <- 2 * sum((dATP*plasmid.A), (dCTP*plasmid.C), (dGTP*plasmid.G), (dTTP*plasmid.T))
chrom.synth.P <- 2 * sum((dATP*chrom.A), (dCTP*chrom.C), (dGTP*chrom.G), (dTTP*chrom.T))
genome.synth.P <- plasmid.synth.P + chrom.synth.P
# plasmid is ~2 % of total genome cost
```

# 2) Cost of helix unwinding

```{r}
# Mostly helicase activity
plasmid.unwind <- plasmid.genome * 1 # where 1 = one P?
chrom.unwind <- chrom.genome * 1
genome.unwind <- plasmid.unwind + chrom.unwind
# unwinding is ~1 % of nucleotide synthesis cost
```

# 3) Chain elongation

```{r}
# Costs associated with annealing of Okazaki fragements on lagging strands of synthesis
Lp <- 10 # RNA primer length, bp
Lof <- 1500 # mean length of Okazaki fragment
rib.P <- 48 # number of ATP equivalents per ribonucleotide
Okazaki.cost.plasmid <- (rib.P * Lp * plasmid.genome) / Lof
Okazaki.cost.chrom <- (rib.P * Lp * chrom.genome) / Lof
Okazaki.cost <- Okazaki.cost.plasmid + Okazaki.cost.chrom
# Chain elongation ~0.3 % of nucleotide synthesis cost
```

#4) Post-polymerization modification and DNA repair

```{r}
# Poor estimates, but thought to be minor relative to other costs
# Ignored in this study
```

# Total costs

```{r}
# Total cost of a cell at the gene level for Bacillus, which includes
# genes needed for vegetative growth and sporulation

total.gene.cost <- genome.synth.P + genome.unwind + Okazaki.cost
# 4.3 x 10^7 P

# Could conceivably pull out all the genes required for sporulation
# Galperin suggests ther are on the order of 200
# Could separate those out to quantify the gene-level burden of sporulation
# Effectively would require just the some of concatenated genes
# Maybe just assume a dNTP cost of 50
```

# II. Cost at the transcriptional level following Lynch and Marinov (2015)

# 1) Costs of ribonucleotide synthesis

```{r}
# Load data

# First, get lengths of genes in B. subtilis: http://www.subtiwiki.uni-goettingen.de/v3/gene/batch (Lr,mat)
bs.gene.lengths <- read.table("data/transcription/gene.length.txt", sep = "\t", header=TRUE)

# Second, get expression data here: http://sporeweb.molgenrug.nl/
veg.transcript <- read.table("data/transcription/1.vegetative.txt", sep = "\t", header=TRUE)

# Merge data sets based on "Genes""
veg <- merge(x = veg.transcript, y = bs.gene.lengths, by.x = "Gene", by.y = "Gene")

# Check merger with "LocusTag"
veg$LocusTag.x = veg$LocusTag.y

# Take average expression for each gene (row) from t = 0 - t = 8
mean.veg <- apply(veg[5:12], 1, mean, na.rm = TRUE)

# Multiply average expression by gene length
veg.synth.gene <- mean.veg * veg$Length

Ribo.cost <- 48 # P units

Ribo.cost.total <- Ribo.cost * sum(veg.synth.gene)

# 2.3 x 10^7 P during vegetative state

```